umbraco/Umbraco-CMS
# Adds new dotnet new template arg --use-https-redirect

Adds new feature to dotnet new template that was required by the Cloud team, as UmbracoID when cloned down locally from Cloud requires the redirect flow to be running on HTTPS and not HTTP

This new flag adds the following to startup.cs
`app.UseHttpsRedirection();`

## Docs from MS Templating Repo for dotnet new and conditionals in .CS files
https://github.com/dotnet/templating/wiki/Conditional-processing-and-comment-syntax#language-source-files

----

To Test (at least this is how I did it):

- Check out the PR and amend your local `src\Directory.Build.props` file to create a new version, e.g. `<InformationalVersion>9.0.0-rc003abc</InformationalVersion>`
- Run `build\build.ps`
- Copy the resulting `Umbraco.Templates.9.0.0-rc003abc.nupkg` from `build.out` into a local folder set up as a NuGet feed
- Run:
- Verify that the Startup.cs class does NOT include `app.UseHttpsRedirection();` 
- Ensure browsing to site on HTTP does NOT auto redirect
```
dotnet new -i Umbraco.Templates::9.0.0-rc003abc
dotnet new umbraco --name NoRedirect
cd NoRedirect
code .
dotnet run
```

- Retest with the following and verify  `app.UseHttpsRedirection();`  is in Startup.cs
- Verify that the Startup.cs class DOES include `app.UseHttpsRedirection();` 
- Ensure browsing to site on HTTP DOES auto redirect to HTTPS

```
dotnet new umbraco --name WithRedirect --use-https-redirect
cd WithRedirect
code .
dotnet run
```
